controllers/clusterversion_controller.go

```diff
@@ -604,6 +624,63 @@ func labelClientOperatorSubscription(c *ClusterVersionReconciler) error {
 		}
 	}
 
+	c.subscriptionChannel = sub.Spec.Channel
+
 	c.log.Info("successfully labelled ocs-client-operator subscription")
 	return nil
 }
+
+func (c *ClusterVersionReconciler) reconcileSubscription() error {
+
+	storageClients := &v1alpha1.StorageClientList{}
+	if err := c.list(storageClients); err != nil {
+		return fmt.Errorf("failed to list storageclients: %v", err)
+	}
+
+	var desiredChannel string
+	for idx := range storageClients.Items {
+		storageClient := &storageClients.Items[idx]
+		annotations := storageClient.GetAnnotations()
+		if annotatedChannel, ok := annotations[utils.DesiredSubscriptionChannelAnnotationKey]; ok {
+			if desiredChannel != "" && desiredChannel != annotatedChannel {
+				// not all storageclients desiring to be on same channel
+				return nil
+			}
+			desiredChannel = annotatedChannel
+		}
+	}
+
+	if desiredChannel == "" {
+		// no storageclients mention a desired channel, user can bump the channel manually if required
+		return nil
+	}
+
+	subscriptions := &opv1a1.SubscriptionList{}
+	// we are already setting this label during reconcile phases
+	err := c.list(subscriptions, client.InNamespace(c.OperatorNamespace), client.MatchingLabels{subscriptionLabelKey: subscriptionLabelValue})
+	if err != nil {
+		return fmt.Errorf("failed to list subscriptions with known labels: %v", err)
+	}
+	if len(subscriptions.Items) != 1 {
+		return fmt.Errorf("failed to validate the presence and uniqueness of ocs-client-operator subscription")
+	}
```

Comment:
Why check and fail for that? what does it give us?

Reply:
we want to ensure there's only one subscription for ocs-client-operator w/ the label set, if there are more than one, something from outside changed it and so failing to update it.

Reply:
What? OCS can only be deployed once on a cluster. This is a basic assumption. You don't need to test for it

Reply:
done.